ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLd3d.cpp

```diff
@@ -66,7 +66,8 @@ void graphics_state_flush_fog() {
 
 void graphics_state_flush_lighting() {
   glShadeModel(d3dShading?GL_SMOOTH:GL_FLAT);
-  glLightModelfv(GL_LIGHT_MODEL_AMBIENT, d3dLightingAmbient);
+  const float glAmbientColor[4] = {COL_GET_Rf(d3dLightingAmbient),COL_GET_Gf(d3dLightingAmbient),COL_GET_Bf(d3dLightingAmbient),1.0f};
```

Comment:
How many times is the lighting flushed per frame? Better to cache this instead of doing four float divides per flush.

Reply:
We flush the render state whenever you start a new a batch and the render state is dirty. We also flush the render state if the user submits a vertex buffer.